### PR TITLE
Фролова Елизавета. Вариант 27. Технология TBB. Выделение ребер на изображении с использованием оператора Собеля.

### PR DESCRIPTION
**Pull request by ElizavetaFrolova**
Программа реализует фильтр Собеля для выделения границ на изображении с использованием параллельной обработки через Intel TBB.

Сначала изображение преобразуется из цветного (RGB) в оттенки серого, затем применяется оператор Собеля для вычисления градиентов яркости по горизонтали и вертикали. Итогом является изображение с выделенными контурами.

Параллелизм реализован в двух местах:
1. При преобразовании изображения в серый (по пикселям).
2. При применении фильтра Собеля (по строкам и столбцам изображения).

### CODE REVIEW

File: tasks/tbb/frolova_e_Sobel_filter/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+      int gradient = static_cast<int>(std::sqrt((res_x * res_x) + (res_y * res_y)));
+      result[(y * width) + x] = std::clamp(gradient, 0, 255);
+    }
+  }
+
+  return result;
+}
+
+}  // namespace
+
```
Comment (wabka22) [inline tasks/tbb/frolova_e_Sobel_filter/func_tests/main.cpp:96]: please add a test with an image consisting of zero values.

**Review by wabka22 — APPROVED** [review]

File: tasks/tbb/frolova_e_Sobel_filter/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+  task_data->inputs.emplace_back(reinterpret_cast<uint8_t *>(pict.data()));
+  task_data->inputs_count.emplace_back(pict.size());
+
+  task_data->outputs.emplace_back(reinterpret_cast<uint8_t *>(res.data()));
+  task_data->outputs_count.emplace_back(res.size());
+
+  // Create Task
+  frolova_e_sobel_filter_tbb::SobelFilterTBB test_task(task_data);
+  ASSERT_EQ(test_task.Validation(), false);
+}
```
Comment (MaxikGuy) [inline tasks/tbb/frolova_e_Sobel_filter/func_tests/main.cpp:357]: Please add tests for a completely black and completely white image.

**Review by MaxikGuy — APPROVED** [review]

**Review by allnes — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]
